trpcultivate_phenotypes/tests/src/Unit/ConfigWatermarkFormTest.php

```diff
@@ -21,103 +21,136 @@
+   */
+  public function testBuildForm() {
     $form = [];
     $form_state = new FormState();
-    $config_form = $watermark->buildForm($form, $form_state);
+
+    $config_form = $this->watermark_form->buildForm($form, $form_state);
 
     // Form theme is system configuration type.
-    $this->assertEquals('system_config_form', $config_form['#theme']);
-    // Field types.
-    $this->assertEquals('radios', $config_form['charts']['#type']);
+    $this->assertEquals(
+      'system_config_form',
+      $config_form['#theme'],
+      'The theme of the form generated by buildForm() method does not match expected theme.'
+    );
+
+    // Form has radio button field.
+    $this->assertEquals(
+      'radios',
+      $config_form['charts']['#type'],
+      'The form generated by buildForm() method is missing a form field of type radio button.',
+    );
   }
 
   /**
-   * Test validate functionality of WatermarkForm class.
+   * Test validateForm() method.
    */
   public function testValidateForm() {
-    $watermark = \Drupal::service('watermark.config');
-
-    // Test if it is the watermark config form using the form id.
-    $this->assertEquals('trpcultivate_phenotypes_watermark_settings_form', $watermark->getFormId());
-
-    // Method formValidate requires 2 parameters $form and $form_state.
-    $form_state = new FormState();
     $form = [];
+    $form_state = new FormState();
 
     // Validate that when choosing to watermark, an image file is required.
     $form_state->setValue('charts', 1);
     $form_state->setValue('file', []);
-    $watermark->validateForm($form, $form_state);
-    $this->assertTrue($form_state->hasAnyErrors());
+    $this->watermark_form->validateForm($form, $form_state);
+    $this->assertTrue(
+      $form_state->hasAnyErrors(),
+      'If watermark charts was set to true, then the formValidate() method is expected to trigger an error if no watermark image was provided.'
+    );
 
     $form_state->clearErrors();
 
-    // Validate that when choosing not to watermark,
-    // an image file is not required.
+    // Validate that when choosing not to watermark, image file is not required.
     $form_state->setValue('charts', 0);
     $form_state->setValue('file', []);
-    $watermark->validateForm($form, $form_state);
-    $this->assertFalse($form_state->hasAnyErrors());
+    $this->watermark_form->validateForm($form, $form_state);
+    $this->assertFalse(
+      $form_state->hasAnyErrors(),
+      'If watermark charts was set to false, then the formValidate() method is expected to submit the form without a watermark image.'
+    );
+  }
+
+  /**
+   * Test submitForm() method.
+   *
+   * @todo this test produces an error: Call to member function set() on null.
+   */
+  public function testSubmitForm() {
+    $form = [];
+    $form_state = new FormState();
+
+    $form_state->setValue('charts', 0);
+    $form_state->setValue('file', $form);
+
+    // $this->watermark_form->submitForm($form, $form_state);
   }
 
```

Comment:
This is considered a risky test since it doesn't have any assertions. We should simply remove it since this is tested in a kernel test.

```suggestion

```